### PR TITLE
chore: Update compatible node version

### PR DESCRIPTION
**Pull request by w3nl**
## Proposed Changes

Allow Node 20.x

## Description

Now you allow only Node 20.0.0, let's make it compatible with all Node 20.x.x versions.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "cloudevents",
-  "version": "7.0.0",
+  "version": "7.0.1",
```
Comment (lance) [inline package.json:3]: This should stay at 7.0.0
```suggestion
  "version": "7.0.0",
```

Comment (pieter-pon) [reply package.json]: I've changed it back

Comment (lance) [reply package.json]: @pieter-pon thanks - you also should revert package-lock.json.

Comment (pieter-pon) [reply package.json]: @lance Also revert the package-lock, should I update the lock file that only the change for the engine is in the lock file, just like the normal package.json?

Comment (w3nl) [reply package.json]: @lance Also revert the package-lock, should I update the lock file that only the change for the engine is in the lock file, just like the normal package.json?

**Review by lholmquist — APPROVED** [review]

**Pull request merged**